models/image.py

```diff
@@ -1,10 +1,133 @@
 
-from models.base import Base
+import sqlalchemy as sa
+from sqlalchemy import orm
+from sqlalchemy.dialects.postgresql import JSONB
 
+from models.base import Base, FileOnDiskMixin, SpatiallyIndexed
+
+
+class Image(Base, FileOnDiskMixin, SpatiallyIndexed):
 
-class Image(Base):
     __tablename__ = 'images'
 
+    exposure_id = sa.Column(
+        sa.ForeignKey('exposures.id'),
+        nullable=True,
+        index=True,
+        doc=(
+            "ID of the exposure from which this image was derived. "
+            "Only set for single-image objects."
+        )
+    )
+
+    exposure = orm.relationship(
+        'Exposure',
+        cascade='save-update, merge, refresh-expire, expunge',
+        doc=(
+            "Exposure from which this image was derived. "
+            "Only set for single-image objects."
+        )
+    )
+
+    source_images = orm.relationship(
+        'Image',
+        cascade='save-update, merge, refresh-expire, expunge',
+        doc=(
+            "Images used to produce a multi-image object "
+            "(e.g., an images stack, reference, difference, super-flat, etc)."
+        )
+    )
+
+    is_multi_image = sa.Column(
+        sa.Boolean,
+        nullable=False,
+        default=False,
+        index=True,
+        doc=(
+            "Whether this image is a multi-image object. "
+            "Multi-image objects will have a null exposure, "
+            "and a list of Image objects from which it was derived. "
+        )
+    )
+
+    combine_method = sa.Column(
+        sa.Text,
+        nullable=True,
+        index=True,
+        doc=(
+            "Type of combination used to produce this multi-image object. "
+            "One of: coadd, subtraction. "
+        )
+    )
+
+    type = sa.Column(
+        sa.Text,
+        nullable=False,
+        index=True,
+        doc=(
+            "Type of image. One of: science, reference, difference, bias, dark, flat. "
+        )
+    )
```

Comment:
Again, if we're predefining the types, I'd make it an ENUM.

Reply:
ok